extension/js/common/api/key-server/attester.ts

```diff
@@ -25,7 +25,34 @@ export class Attester extends Api {
       console.info(`Skipping attester lookup of ${email} because attester search on this domain is disabled.`);
       return { pubkeys: [] };
     }
-    return await this.doLookup(email);
+    // Can't use Promise.allSettled because we are using ES2018 (Available from ES2020)
+    const results = await promiseAllSettled([
+      this.doLookupLdap(email),  // get from recipient-specific LDAP server, if any, relayed through flowcrypt.com
+      this.doLookup(email),  // get from flowcrypt.com public keyserver database
+      this.doLookupLdap(email, 'keyserver.pgp.com'), // get from keyserver.pgp.com, relayed through flowcrypt.com
+    ]);
+    const validResults = results.filter(result => result.status === 'fulfilled' && result.value);
+    for (const result of validResults) {
+      if (result.value!.pubkeys.length) {
+        return result.value!;
+      }
+    }
```

Comment:
If there was an error searching `flowcrypt.com` (non-ldap) and no key was found elsewhere, than that error should be surfaced (re-thrown). Including network errors, 500s, timeouts etc. We expect that endpoint to respond normally.

I don't want to treat responses from `flowcrypt.com/attester/pub/xxx` as not-found when the server cannot be reached or returns error.

I think already on current codebase, doing that should cause a red color recipient that could be retried.

Reply:
It's odd we don't have that scenario tested.

Please add a test `allows to retry public key search when attester returns error` that would search `attester.return.error@flowcrypt.test` and add `attester-endpoints` code that will return 500 for that address (or find existing email that already does)

Searching that in secure compose should render it as search error, instead of a no-pubkey result.